spec/active_model/validations/store_model_validator_spec.rb

```diff
@@ -80,24 +80,134 @@
+          expect(subject.configuration.second.errors.full_messages).to be_empty
+        end
+      end
+
+      context "when some array members are invalid" do
+        let(:attributes) { { configuration: [Configuration.new(color: "red"), Configuration.new] } }
+
+        it { is_expected.to be_invalid }
+
+        it "returns correct error messages" do
+          expect(subject.errors.messages).to eq(configuration: ["[1] is invalid"])
+          expect(subject.errors.full_messages).to eq(["Configuration [1] is invalid"])
+
+          expect(subject.configuration.second.errors.messages).to eq(color: ["can't be blank"])
+          expect(subject.configuration.second.errors.full_messages).to eq(["Color can't be blank"])
+        end
       end
 
-      context "when array member is invalid" do
-        let(:attributes) { { configuration: [Configuration.new] } }
+      context "when more than one array member is invalid" do
+        let(:attributes) { { configuration: [Configuration.new, Configuration.new] } }
 
         it { is_expected.to be_invalid }
 
         it "returns correct error messages" do
-          expect(subject.errors.messages).to eq(configuration: ["is invalid"])
-          expect(subject.errors.full_messages).to eq(["Configuration is invalid"])
+          expect(subject.errors.messages).to eq(configuration: ["[0] is invalid", "[1] is invalid"])
+          expect(subject.errors.full_messages).to eq(
+            [
+              "Configuration [0] is invalid",
+              "Configuration [1] is invalid"
+            ]
+          )
 
           expect(subject.configuration.first.errors.messages).to eq(color: ["can't be blank"])
           expect(subject.configuration.first.errors.full_messages).to eq(["Color can't be blank"])
+
+          expect(subject.configuration.second.errors.messages).to eq(color: ["can't be blank"])
+          expect(subject.configuration.second.errors.full_messages).to eq(["Color can't be blank"])
         end
       end
 
-      context "when store_model value is nil" do
+      context "with merge_errors: true" do
+        let(:custom_product_class) do
+          build_custom_product_class do
+            attribute :configuration, Configuration.to_array_type
+            validates :configuration, store_model: { merge_errors: true }
+          end
+        end
+
+        context "when more than one array member is invalid" do
+          let(:attributes) { { configuration: [Configuration.new(color: "red"), Configuration.new] } }
+
+          it { is_expected.to be_invalid }
+
+          it "returns correct error messages" do
+            if Rails::VERSION::MAJOR < 6 || Rails::VERSION::MAJOR == 6 && Rails::VERSION::MINOR == 0
+              # this feels really weird what do we want the behavior to be in this case?
+              # Seems like merging errors in the array case is kinda broken without the Rails 6 pattern
+              # Maybe we should always do the Rails 6 pattern in the array case
```

Comment:
Thoughts? 

The other option would be to change the key ie make it `color[0]` or something but why given Rails 6.1 is the future.